lib/proxy/log.rb

```diff
@@ -12,79 +9,49 @@ def logger
   end
 
   class LoggerFactory
-    class LogFormatter < ::Logger::Formatter
-      Format = "%s, [%s%s] %5s -- %s: %s\n".freeze
-
-      def call(severity, time, progname, msg)
-        Format % [severity[0..0], format_datetime(time), request_id[0..7], severity, progname, msg2str(msg)]
-      end
-
-      def request_id
-        Thread.current.thread_variable_get(:request_id).to_s
-      end
-    end
-
     begin
       require 'syslog/logger'
-      ::Syslog::Logger.class_eval { alias_method :write, :info }
-
-      class SyslogFormatter < Syslog::Logger::Formatter
-        Format = "<%s> %s\n".freeze
-
-        def call(severity, time, progname, msg)
-          request_id.empty? ? clean(msg) : Format % [request_id[0..7], clean(msg)]
-        end
-
-        def request_id
-          Thread.current.thread_variable_get(:request_id).to_s
-        end
-      end
-
       @syslog_available = true
     rescue LoadError
-      # ignore, syslog isn't available on this platform
       @syslog_available = false
     end
 
     def self.logger
+      logger_name = 'foreman-proxy'
+      layout = Logging::Layouts.pattern(pattern: "%d %.8X{request} [%.1l] %m\n")
+      notime_layout = Logging::Layouts.pattern(pattern: "%.8X{request} [%.1l] %m\n")
+      logger = Logging.logger.root
       if log_file.casecmp('STDOUT').zero?
         if SETTINGS.daemon
-          puts "Settings log_file=STDOUT and daemon=true are incompatible, exiting..."
+          puts "Settings log_file=STDOUT and daemon=true cannot be used together"
           exit(1)
         end
-        logger = stdout_logger
+        logger.add_appenders(Logging.appenders.stdout(logger_name, layout: layout))
       elsif log_file.casecmp('SYSLOG').zero?
         unless syslog_available?
-          puts "'SYSLOG' logger is not supported on this platform, please use STDOUT or a file-based logger. Exiting..."
+          puts "Syslog is not supported on this platform, use STDOUT or a file"
           exit(1)
         end
-        logger = syslog_logger
+        logger.add_appenders(Logging.appenders.syslog(
+          logger_name, layout: notime_layout, facility: ::Syslog::Constants::LOG_LOCAL5))
+      elsif log_file.casecmp('JOURNAL').zero? || log_file.casecmp('JOURNALD').zero?
+        begin
+          logger.add_appenders(Logging.appenders.journald(
+            logger_name, logger_name: :proxy_logger, layout: notime_layout, facility: ::Syslog::Constants::LOG_LOCAL5))
+        rescue NoMethodError
+          logger.add_appenders(Logging.appenders.stdout(logger_name, layout: layout))
+          logger.warn "Journald is not available on this platform. Falling back to STDOUT."
+        end
       else
-        logger = default_logger(log_file)
+        begin
+          logger.add_appenders(Logging.appenders.rolling_file(
+            logger_name, layout: layout, filename: log_file, keep: 6, size: 1024*1024*100, age: 'weekly', roll_by: 'date'))
```

Comment:
My own preferences is to use in such cases a constant for the size.
And also this part can be a configurable settings (the rotation part that is).

What do you think on that points?

Reply:
Oh do you mean constant like `SIZE = 1024*1024*100`? It's not needed, this is called only once as this is a singleton, therefore there is no performance benefit at all (technically it's even slower).

Reply:
Anyway this is now configurable and irrelevant :-)